src/deroff.rs

```diff
@@ -1028,19 +1044,107 @@ impl Deroffer {
+                        let mut option = self.s.to_owned();
+                        let mut arg = String::new();
+
+                        let mut idx: usize = 0;
+
+                        while option
+                            .get(idx..=idx)
+                            .and_then(|s| Some(s.chars().all(|c| c.is_alphabetic())))
+                            .unwrap_or_default()
+                        {
+                            idx += 1;
+                        }
+
+                        if option.get(idx..=idx) == Some("(") {
+                            option = option.get(..idx).unwrap_or_default().to_owned();
+                            self.s = self.s.get(idx + 1..).unwrap_or_default().to_owned();
+                            arg = self.s.clone();
+                        } else {
+                            self.s = "".to_owned();
+                        }
+
+                        if !arg.is_empty() {
+                            if let Some(idx_) = arg.find("(") {
+                                idx = idx_;
+                                arg = arg.get(..idx).unwrap_or_default().to_owned();
+                            }
+                            self.s = self.s.get(idx + 1..).unwrap_or_default().to_owned();
+                        } else {
+                            // This was commented out in the python, so it is here too :)
+                            // self.skip_char(1);
+                        }
+
+                        if option.to_lowercase() == "tab" {
+                            self.tblTab = arg.get(0..1).unwrap_or_default().to_owned();
+                        }
+                    }
+                }
+                self.tblstate = TblState::Format;
+                self.condputs("\n");
+            }
+            TblState::Format => {
+                while !self.s.is_empty() && !".\n".contains(self.str_at(0)) {
+                    self.skip_leading_whitespace();
+                    if !self.str_at(0).is_empty() {
+                        self.skip_char(1);
+                    }
+                }
+
+                if self.str_at(0) == "." {
+                    self.tblstate = TblState::Data;
+                }
+                self.condputs("\n");
+            }
+            TblState::Data => {
+                if !self.tblTab.is_empty() {
+                    self.s = self.s.replace(&self.tblTab, "\n");
+                }
+
+                self.text();
+            }
+        }
+
+        true
+    }
+
+    fn do_line(&mut self) -> bool {
+        match self.s.chars().nth(0) {
+            Some('.') | Some('\'') => !self.request_or_macro(),
+            Some(_) => {
                 if self.tbl {
-                    do_tbl(s)
+                    self.do_tbl()
                 } else {
-                    text(s)
+                    self.text()
                 }
             }
             None => panic!("do_line` called with empty string as argument"),
```

Comment:
```suggestion
            None => unreachable!(),
```
Or better, we could just use `expect` or `unwrap` since the original code does not handle this. So no need to match `Some`.

Reply:
```rust
match self.s.bytes().next().unwrap_or_else(|| unreachable!()) { .. }
```